<a target="_blank" href="/drive/12wU0lxB2xo00DNRYN3LfFOYMTfZoTcZG#scrollTo=hHJWeFe3vASz">
  <img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/>
</a>

# Importação de bibliotecas globais

In [1]:
import pandas

###<strong>Pandas</strong>
` O Pandas é uma biblioteca de código aberto em Python amplamente utilizada para análise e manipulação de dados. Ele fornece estruturas de dados eficientes e fáceis de usar, como o DataFrame, que permite armazenar e manipular dados tabulares de forma flexível. O Pandas também oferece uma variedade de funcionalidades para limpeza, transformação e análise de dados, incluindo operações de indexação, filtragem, agregação e visualização. Com o Pandas, os usuários podem carregar dados de várias fontes, como arquivos CSV, Excel, SQL, e realizar tarefas como seleção, junção, agrupamento e cálculos estatísticos de forma simples e eficiente. `

#Faculdades Publicas <hr />

In [3]:
arquivoFacPublicas = pandas.read_csv("/content/drive/MyDrive/Projeto Teste/Projeto Final/DADOS/Cópia de faculdades_publicas.csv")
df_faculdades_publicas = pandas.DataFrame(arquivoFacPublicas)

> Modelagem dados

In [4]:
media_total = df_faculdades_publicas['brancos '].mean() + df_faculdades_publicas['negros '].mean() + df_faculdades_publicas['pardos'].mean()
total_alunos_eng = df_faculdades_publicas['brancos '].max() + df_faculdades_publicas['negros '].max() + df_faculdades_publicas['pardos'].max()
porcentagem_brancos = (df_faculdades_publicas['brancos '].mean() / media_total) *100
porcentagem_negros = (df_faculdades_publicas['negros '].mean() / media_total) *100
porcentagem_pardos = (df_faculdades_publicas['pardos'].mean() / media_total) *100

> Exibição dos dados

In [5]:
print('\n----------- Descrição estatistica pandas ----------')
print(df_faculdades_publicas.head())


print('\n----------- Maximo valor ----------')
print(df_faculdades_publicas.max())

print('\n----------- Relação entre valores ----------')
print(f'=> Total de pessoas, de qualquer raça, matriculadas no curso de engenharia e construção : "{round(total_alunos_eng)}"')
print(f'=> A media de pessoas, de qualquer raça, ingressadas na faculdade, em qualquer curso, é: "{round(media_total)}" pessoas')
print(f'=> A porcentagem de brancos é de: {round(porcentagem_brancos,2)}%')
print(f'=> A porcentagem de negros é de: {round(porcentagem_negros,2)}%')
print(f'=> A porcentagem de pardos é de: {round(porcentagem_pardos,2)}%')


----------- Descrição estatistica pandas ----------
                          brancos   negros   pardos
negócios e administração     11044     2634    8155
direito                      36772     9802   24277
ciencias sociais              7642     2088    4826
ciencias exatas               5843     1200    3587
ciencias naturais             3622      803    2686

----------- Maximo valor ----------
brancos     160269
negros       24846
pardos       94731
dtype: int64

----------- Relação entre valores ----------
=> Total de pessoas, de qualquer raça, matriculadas no curso de engenharia e construção : "279846"
=> A media de pessoas, de qualquer raça, ingressadas na faculdade, em qualquer curso, é: "86448" pessoas
=> A porcentagem de brancos é de: 54.86%
=> A porcentagem de negros é de: 10.22%
=> A porcentagem de pardos é de: 34.92%


##Conclusão previa:
`Neste cenario, é feito uma analise sobre os alunos em faculdades publicas, de tal maneira a explorar possiveis discrepancias em relação a dificuldade de acesso a uma instituição privada por parte do negro em relação ao branco.`

`Com os dados, foram evidenciados a quantidade de alunos matriculados por cor e raça. Foram listados 8 cursos diversos para realizar o estudo`

`Foi feito uma media sobre a quantidade de alunos matriculados, em qualquer curso, levando em consideração cor e raça. O resultado obtido foi uma media de: 55.235 alunos brancos matriculados, 7.653 alunos negros matriculados e 34.493 alunos pardos matriculados;`

####Isto resulta em uma media de:
  >86448 pessoas matriculadas por curso

####Sendo desta quantidade,a taxa de alunos matriculas por cor ou raça
  >Alunos brancos matriculados de 54,86%
  
  >Alunos negros matriculados de 10,22% 

  >Alunos pardos matriculados de 34.92%

`Ou seja, apenas aproximadamente 10% dos alunos matriculados são negros, e mesmo com a miscigenação racial, não chega a equiparar com a quantidade de alunos brancos com diretos de acesso a uma faculdade privada`

`Destes, o curso com maior adesão foi de direito, com 279.846 pessoas no total, sendo que mais da metade dos alunos são brancos e pouco mais de 10% são negros, reforçando ainda mais a tese.`

#Faculdades Particulares <hr />

> Importando dados

In [6]:
arquivo = pandas.read_csv("/content/drive/MyDrive/Projeto Teste/Projeto Final/DADOS/Cópia de faculdades_particulares.csv", sep=",")
df_faculdades_particulares = pandas.DataFrame(arquivo)


> Modelagem dados


In [7]:
media_total = df_faculdades_particulares['brancos '].mean() + df_faculdades_particulares['negros '].mean() + df_faculdades_particulares['pardos'].mean()
total_alunos_direito = df_faculdades_particulares['brancos '].max() + df_faculdades_particulares['negros '].max() + df_faculdades_particulares['pardos'].max()
porcentagem_brancos = (df_faculdades_particulares['brancos '].mean() / media_total) *100
porcentagem_negros = (df_faculdades_particulares['negros '].mean() / media_total) *100
porcentagem_pardos = (df_faculdades_particulares['pardos'].mean() / media_total) *100

> Exibição dos dados

In [8]:
print('\n----------- Descrição estatistica pandas ----------')
print(df_faculdades_particulares.head())


print('\n----------- Maximo valor ----------')
print(df_faculdades_particulares.max())

print('\n----------- Relação entre valores ----------')
print(f'=> Total de pessoas, de qualquer raça, matriculadas no curso de direito: "{round(total_alunos_direito)}"')
print(f'=> A media de pessoas, de qualquer raça, ingressadas na faculdade, em qualquer curso, é: "{round(media_total)}" pessoas')
print(f'=> A porcentagem de brancos é de: {round(porcentagem_brancos,2)}%')
print(f'=> A porcentagem de negros é de: {round(porcentagem_negros,2)}%')
print(f'=> A porcentagem de pardos é de: {round(porcentagem_pardos,2)}%')


----------- Descrição estatistica pandas ----------
                          brancos   negros   pardos
negócios e administração     32611     4765   20488
direito                     310057    43512  196703
ciencias sociais             29347     3720   14992
ciencias exatas                588       53     240
ciencias naturais             1960      210     874

----------- Maximo valor ----------
brancos     310057
negros       43512
pardos      196703
dtype: int64

----------- Relação entre valores ----------
=> Total de pessoas, de qualquer raça, matriculadas no curso de direito: "550272"
=> A media de pessoas, de qualquer raça, ingressadas na faculdade, em qualquer curso, é: "97382" pessoas
=> A porcentagem de brancos é de: 56.72%
=> A porcentagem de negros é de: 7.86%
=> A porcentagem de pardos é de: 35.42%


##Conclusão previa:
`Neste cenario, é feito uma analise sobre os alunos em faculdades particulares, de tal maneira a explorar possiveis discrepancias em relação a dificuldade de acesso a uma instituição privada por parte do negro em relação ao branco.`

`Com os dados, foram evidenciados a quantidade de alunos matriculados por cor e raça. Foram listados 8 cursos diversos para realizar o estudo`

`Foi feito uma media sobre a quantidade de alunos matriculados, em qualquer curso, levando em consideração cor e raça. O resultado obtido foi uma media de: 55.235 alunos brancos matriculados, 7.653 alunos negros matriculados e 34.493 alunos pardos matriculados;`

####Isto resulta em uma media de:
  >97382 pessoas matriculadas por curso

####Sendo desta quantidade,a taxa de alunos matriculas por cor ou raça
  >Alunos brancos matriculados de 56,72%
  
  >Alunos negros matriculados de 7,86% 

  >Alunos pardos matriculados de 35.42%

`Ou seja, nem 10% dos alunos matriculados são negros, e mesmo com a miscigenação racial, não chega a equiparar com a quantidade de alunos brancos com diretos de acesso a uma faculdade privada`

`Destes, o curso com maior adesão foi de direito, com 550.272 pessoas no total, sendo que mais da metade dos alunos são brancos e pouco menos de 10% são negros, reforçando ainda mais a tese.`

#Ocupações Formais <hr />


> Importação dos arquivos

In [9]:
arquivo_ocupacoes_formais = pandas.read_csv('/content/drive/MyDrive/Projeto Teste/Projeto Final/DADOS/Cópia de ocupacoesfi_renovado.csv', sep=",")
df_ocupacoes_formais = pandas.DataFrame(arquivo_ocupacoes_formais)

> Modelagem de dados:

In [10]:
taxa_pessoas_negras = list((df_ocupacoes_formais['preta'] / df_ocupacoes_formais['total']) * 100)
taxa_pessoas_brancas = list((df_ocupacoes_formais['branca'] / df_ocupacoes_formais['total']) * 100)
taxa_pessoas_pardas = list((df_ocupacoes_formais['parda'] / df_ocupacoes_formais['total']) * 100)

> Exibição dos dados

In [12]:
print('\n----------- Descrição estatistica pandas ----------')
print(df_ocupacoes_formais.head())

print('\n----------- Maximo valor ----------')
print(df_ocupacoes_formais.max())

print('\n----------- Relação entre valores ----------')
print(f'Taxa de pessoas negras em ocupações formais: {round(taxa_pessoas_negras[0],2)}%')
print(f'Taxa de pessoas brancas em ocupações formais: {round(taxa_pessoas_brancas[0],2)}%')
print(f'Taxa de pessoas pardas em ocupações formais: {round(taxa_pessoas_pardas[0],2)}%')

******** Ocupações Formais por cor ou raça no Brasil em 2021 ********
    total  branca  preta   parda
0  53.619  27.221  5.035  20.788

----------- Maximo valor ----------
total     53.619
branca    27.221
preta      5.035
parda     20.788
dtype: float64

----------- Relação entre valores ----------
Taxa de pessoas negras em ocupações formais: 9.39%
Taxa de pessoas brancas em ocupações formais: 50.77%
Taxa de pessoas pardas em ocupações formais: 38.77%


##Conclusão Previa:
`Neste cenario temos um dado nacional, onde temos as relações das pessoas que estão com ocupações de trabalho CLT por cor ou raça. Neste caso foi realizado uma amostragem com 53,619 mil pessoas`

`Nele é possivel constatar que menos de 10% dessas pessoas que possuem uma ocupação formal de trabalho no pais, são negros, sendo composto essa taxa majoritariamente por brancos. Os dados brutos podem ser evidenciados abaixo: `

>Percentual de pessoas negras em ocupações formais: 9.39%

>Percentual de pessoas brancas em ocupações formais: 50.77%

>Percentual de pessoas pardas em ocupações formais: 38.77%

#Cargos Gerenciais <hr />

> Importação de arquivos

In [13]:
arquivo_ocupacoes_gerenciais = pandas.read_csv('/content/drive/MyDrive/Projeto Teste/mercado de trabalho e educação/csv/cargoscorouraça_renovado.csv', sep=",")
df_ocupacoes_gerenciais = pandas.DataFrame(arquivo_ocupacoes_gerenciais)

> Modelagem dos dados

In [14]:
taxa_pessoas_negras = list(df_ocupacoes_gerenciais['preta '])
taxa_pessoas_brancas = list(df_ocupacoes_gerenciais['branca'])
taxa_pessoas_pardas = list(df_ocupacoes_gerenciais['parda'])

> Exibição dos dados

In [15]:
print('\n----------- Descrição estatistica pandas ----------')
print(df_ocupacoes_gerenciais.head())

print('\n----------- Maximo valor ----------')
print(df_ocupacoes_gerenciais.max())

print('\n----------- Relação entre valores ----------')
print(f'Taxa de pessoas negras em cargos gerenciais: {taxa_pessoas_negras[0]}%')
print(f'Taxa de pessoas brancas em cargos gerenciais: {taxa_pessoas_brancas[0]}%')
print(f'Taxa de pessoas pardas em cargos gerenciais: {taxa_pessoas_pardas[0]}%')


----------- Descrição estatistica pandas ----------
   total  branca  preta   parda
0   2387    69.0     4.3   25.3

----------- Maximo valor ----------
total     2387.0
branca      69.0
preta        4.3
parda       25.3
dtype: float64

----------- Relação entre valores ----------
Taxa de pessoas negras em cargos gerenciais: 4.3%
Taxa de pessoas brancas em cargos gerenciais: 69.0%
Taxa de pessoas pardas em cargos gerenciais: 25.3%


##Conclusão
`Seguindo a ideia acima, podemos fazer uma observação que tambem é relevante neste cenario, seriam os da taxa de trabalhadores que possuem participações em cargos gerenciais dentro de alguma empresa. Pouco menos de 5% dos negros, apenas, tem conquistado ou possuem cargos elevados dentro de uma empresa, os dados em % podem ser evidenciados abaixo:`
>Percentual de pessoas negras em cargos gerenciais: 4.30%

>Percentual de pessoas brancas em cargos gerenciais: 69.00%

>Percentual de pessoas pardas em cargos gerenciais: 25.30%


`Podemos, portanto salientar que é um cenario muito desigual, no qual pode possuir motivações por varias frentes como a falta de acesso a educação pelos mesmos, o que acaba impedindo-as de conseguir ou alavancar sua carreira para uma cargo mais elevado. Ou, para aqueles que não estão com ocupação formal, ate mesmo levar estas pessoas a informalidade ou a meios ilicitos de se manter`